packages/app-desktop/gui/NoteEditor/utils/contextMenuUtils.ts

```diff
@@ -0,0 +1,85 @@
+import Resource from '@joplin/lib/models/Resource';
+
+export enum ContextMenuItemType {
+	None = '',
+	Image = 'image',
+	Resource = 'resource',
+	Text = 'text',
+	Link = 'link',
+}
+
+export interface ContextMenuOptions {
+	itemType: ContextMenuItemType;
+	resourceId: string;
+	mime: string;
+    filename: string;
+	linkToCopy: string;
+	textToCopy: string;
+	htmlToCopy: string;
+	insertContent: Function;
+	isReadOnly?: boolean;
+}
+
+export interface ContextMenuItem {
+	label: string;
+	onAction: Function;
+	isActive: Function;
+}
+
+export interface ContextMenuItems {
+	[key: string]: ContextMenuItem;
+}
+
+export async function resourceInfo(options: ContextMenuOptions): Promise<any> {
+	const resource = options.resourceId ? await Resource.load(options.resourceId) : null;
+	const filePath = resource ? Resource.fullPath(resource) : null;
+	const filename = resource ? (resource.filename ? resource.filename : resource.title) : options.filename ? options.filename : '';
+	const getCopyPath = () => {
+		if (options.textToCopy && options.mime) {
+			return textToDataUri(options.textToCopy, options.mime);
+		}
+		return filePath;
+	};
+	return { resource, filePath, filename, getCopyPath };
+}
+
+export function textToDataUri(text: string, mime: string): string {
+	return `data:${mime};base64,${Buffer.from(text).toString('base64')}`;
+}
+
+export const svgUriToPng = (document: Document, svg: string) => new Promise<Uint8Array>((resolve, reject) => {
+	let canvas: HTMLCanvasElement;
+	let ctx;
+	const img = document.createElement('img');
+
+	const cleanUpAndReject = () => {
+		if (canvas) canvas.remove();
+		if (img) img.remove();
+		return reject();
```

Comment:
You are losing the error message here - change to `cleanUpAndReject = (error)` then `reject(error)`